gtsam/geometry/Cal3DS2_k3.h

```diff
@@ -0,0 +1,150 @@
+/* ----------------------------------------------------------------------------
+
+ * GTSAM Copyright 2010, Georgia Tech Research Corporation,
+ * Atlanta, Georgia 30332-0415
+ * All Rights Reserved
+ * Authors: Frank Dellaert, et al. (see THANKS for the full author list)
+
+ * See LICENSE for the license information
+
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file Cal3DS2_k3.h
+ * @brief Calibration of a camera with radial distortion whose coefficients are not only k1, k2 but also k3,
+ * calculations in base class Cal3DS2_Base_k3
+ * @date Dec 19, 2023
+ * @author demul
+ * @author Hyeonjin Jeong
+ */
+
+#pragma once
+
+#include <gtsam/geometry/Cal3DS2_k3_Base.h>
+
+namespace gtsam
+{
+/**
+ * @brief Calibration of a camera with radial distortion whose coefficients are not only k1, k2 but also k3
+ * Lie-group behaviors for optimization.
+ * \sa Cal3DS2_Base_k3
+ * @addtogroup geometry
+ * \nosubgrouping
+ */
+class GTSAM_EXPORT Cal3DS2_k3 : public Cal3DS2_k3_Base
+{
+    using Base = Cal3DS2_k3_Base;
+
+   public:
+    enum
+    {
+        dimension = 10
+    };
+
+    /// @name Standard Constructors
+    /// @{
+
+    /// Default Constructor with only unit focal length
+    Cal3DS2_k3() = default;
+
+    Cal3DS2_k3(
+        double fx,
+        double fy,
+        double s,
+        double u0,
+        double v0,
+        double k1,
+        double k2,
+        double p1 = 0.0,
+        double p2 = 0.0,
+        double k3 = 0.0,
```

Comment:
@demul this is some great work, thank you. However, I would recommend adding `k3` to the `Cal3DS2` class and updating the `Cal3DS2_Base` class accordingly. You can ensure backwards compatibility by setting k3 to be 0.0 by default, and group it with k1 and k2.

Reply:
@varunagrawal 
Thanks for your feedback.
I think the way you recommend is better to go eventually.

But I have some points i can't clearly understand.
> You can ensure backwards compatibility by setting k3 to be 0.0 by default, and group it with k1 and k2.
To ensure backwards compatibility, k3 must be placed as a last argument(after tol) and must be 0.0 by default.
But I think placing k3 after tol doesn't look so good.
Actually, that's the point why I created a new class than just updating Cal3DS2.
If you sure just place k3 after tol is fine, I'll refactor as you recommand and request review again.
I'm going to wait your confirmation.